examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,185 @@
+"""
+Example for evolutionary regression on a piecewise target function with one single genome
+=========================================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a piecewise target function.
+Demonstrates the use of if/else branching operator gene.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_piecewise_target_function.py [--max-generations=<N>]
+
+   Options:
+     -h --help
+     --max-generations=<N>  Maximum number of generations [default: 5000]
+"""
+
+import functools
+import matplotlib.pyplot as plt
+import numpy as np
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+
+# %%
+# First we define the new operator node: If/ElseOperator for conditional outputs
+class IfElseOperator(cgp.OperatorNode):
+    """A node that checks its input0 is bigger than 0
+     and outputs input1 if true and input2 otherwise """
```

Comment:
```suggestion
    """A node that outputs the value of its second input if its first input is non-negative, and the value of its third input otherwise."""
```